### ISSUE TITLE
Launch Plan: HogQL

### ISSUE THREAD

**Issue by joethreepwood** [PostHog/meta#103]
## Context
Over the last few weeks, the scope of HogQL has grown and been met with a few (needed, not meant critically) delays. What we originally understood to be just HogQL has grown to include a new Event Explorer, direct SQL access, a beta process, and the need to address some potential security concerns. 

HogQL has been stealth launched, but from a marketing perspective we want to put some resources behind the launch as soon as possible because this launch is both significant in scope and is a powerful USP compared to many competitors. 

At the offsite, [the marketing team discussed what we'd like to see](https://github.com/PostHog/meta/issues/102) from a launch and what we'd need to launch confidently. I took responsibility for coordinating it. So, here we go.  

## Status 
Launch activity for HogQL is currently on pause, as there needs to be some proper pentesting checks run by @lharries first. 

We have [a drafted blog post](https://github.com/PostHog/posthog.com/pull/5469/files), which was all that was originally planned for the launch.

### Launch day activity
- [ ] [Blog post announcement](https://github.com/PostHog/posthog.com/pull/5469/files) - @joethreepwood 
- [ ] [Dedicated blog post art](https://github.com/PostHog/posthog.com/issues/5679) - @lottiecoxon 
- [ ] Include in weekly changelog (short info, link to main announcement) - @joethreepwood 
- [ ] Docs update - Probably @pjhul ?
- [ ] Tutorial following a typical HogQL use case - @ivanagas 
- [ ] In-app banner announcement - @joethreepwood 
- [ ] Social posts on all platforms - @joethreepwood 
- [ ] User Slack announcement - @joethreepwood 
- [ ] ShowHN / HackerNews announcement - @jamesefhawkins

### Launch week activity
- [ ] Paid ads around this new feature & direct SQL access - @joethreepwood to tell Hey Digital
- [ ] Update all competitor comparisons - @joethreepwood 

### Follow-on activity
- [ ] Technical blog post about how we built HogQL (or other topic) - @joethreepwood / @mariusandra 
- [ ] Case study about a user who has adopted this feature - @joethreepwood 

## Requested Feedback
I'd love feedback on all of the above. 

We also need input on when we're likely to be able to launch this activity, so we can create the content in advance. **In a vacuum, I'd propose a launch date of April 17th.** 

That date gives us this week to coordinate, plus most of next week for security testing and any other changes. 

I'd love feedback from people tagged above, plus @andyvan-ph and @charlescook-ph

**Comment by andyvan-ph** [comment]
Looking good. Thoughts from me:

- 17th April won't happen. Marius can provide more context, but my best guess is end of April is more realistic.

- I think we should do a video / trailer for the launch. It's not something we normally do, but would be a good stretch and a useful trial run for PostHog 3000, which we'll definitely want one for.

- Paid ads?

- Given the importance of the launch, I think doing some newsletter sponsorships for HogQL would be worthwhile. Maybe Pointer + TL;DR. Thoughts @charlescook-ph?

**Comment by mariusandra** [comment]
Hey, I'm off from basically tomorrow until the 17th of April (excluding), so we're going to need someone else to finish the missing features, UX improvements, docs, etc, if we want to hit that deadline. End of April sounds more likely. We'll obviously be testing with beta users when I'm off, but I'd leave an extra 2 weeks for the final polish.

The scope of this launch is also making me a bit nervous. Normally when we "launch" features, they're not finished. We slap a "beta" label on a thing, and remove a feature flag. However in this case, if we're launching it with a video trailer and paid ads, this "SQL support" better work damned well out of the box. This sets different expectations on engineering. I also can't predict the timing and results of the pen test. All of it to say: let's keep being flexible :).

To be clear, I like the idea of a big launch. This is a big enough differentiating feature that any promotion for it will attract a lot of curious new users.

**Comment by joethreepwood** [comment]
I'm happy to push back a launch date to whenever works for others - my goal is just to have a hard deadline that we can all coordinate to. Flexibility guaranteed. 

Likewise, happy to hold back paid ads until a few weeks later. 

I'm not _very_ convinced by video trailer. I'm sure we could find a way to make it fun, but I'm a big fuzzy on what the content of it would be or where we'd use it. @andyvan-ph Are you envisaging a paid video ad, a video tutorial, or something else?

**Comment by mariusandra** [comment]
Can we make the video self-ironic: "so what are you working on" - "a trailer to promote the launch of SQL support" - "how corporate of you... wait, how is it that SQL support is something you launch... doesn't everyone just support SQL?" - "nope, turns out it's a big deal" - "odd" - "I know, right" --> cut to some snazzy message about this not being 2004 anymore.

**Comment by joethreepwood** [comment]
Hell, we can make that as a gif. 🤷

**Comment by charlescook-ph** [comment]
Yeah on paid ads and newsletter sponsorships, we usually wait a few weeks til after launch of new features just so we have time to iron out any issues. Once @mariusandra is confident enough, we can pick a specific week to do a blast of paid activity, but it can be in May/June - it's ok if it's separate from the other launch activities. We'd be looking at ~$20k spend here. 

I'm going to be a boring badger and say we shouldn't make a trailer - it feels a bit forced for its own sake and we're not good at video ads. Tutorial content and GIFs showing snippets - 100%, but I think we'll be wasting time and $$$ making a video ad.

**Comment by charlescook-ph** [comment]
(Also generally as a marketing team I'd like to encourage us to not get caught up trying to do big, highly-coordinated launch campaigns - we're not great at them and it's something we've [explicitly said](https://posthog.com/handbook/growth/marketing#things-we-dont-want-to-spend-time-on) we don't want to be good at.)

**Comment by ivanagas** [comment]
A few of ideas:
- Tutorial and tutorial video on some basic HogQL (insights) use cases (already mentioned 👍) 
- An intro to HogQL statements, create a "cheat sheet" like [this](https://learnsql.com/blog/sql-basics-cheat-sheet/)
- Instead of a "trailer," do a video, twitter thread, Slack post(?) answering 10 old Slack questions with HogQL
- An English to HogQL translator using ChatGPT like [this](https://www.autoregex.xyz/)
- Understanding HogQL for non-technical people. Learning the basics of databases with HogQL. 
- A peek behind the scenes of what HogQL is doing

**Comment by joethreepwood** [comment]
> (Also generally as a marketing team I'd like to encourage us to not get caught up trying to do big, highly-coordinated launch campaigns - we're not great at them and it's something we've [explicitly said](https://posthog.com/handbook/growth/marketing#things-we-dont-want-to-spend-time-on) we don't want to be good at.)

Agree as a rule (and especially video-wise) but (just to add context) this was something we discussed at the offsite because the launch plan and scope kept changing, so some level of coordination felt needed.

**Comment by andyvan-ph** [comment]
@jamesefhawkins – We included a ShowHN announcement as part of the original launch plan. You still up for doing this? I figure, if we do, we maybe save it for a couple of days after live just to iron out kinks.
